assets/differentialabundance_report.Rmd

```diff
@@ -871,6 +914,42 @@ if (any(unlist(params[paste0(possible_gene_set_methods, '_run')]))){
 }
 ```
 
+```{r, echo=FALSE, results='asis', eval=params$gprofiler2_run}
+enrichment_files <- grep("gprofiler2", list.files(params$input_dir), value=T, fixed=T)
+cat(paste0("\n### Pathway enrichment analysis {.tabset}"))
```

Comment:
This needs to be within the contrast-wise loop above- you'll see from the structure there that I anticipated multiple gene set methods in future. You should be able to infer the files you'll need to read from the contrast variables available there, rather than inferring contrasts from the gprofiler files.

Reply:
@pinin4fjords I'm wondering if we want to differentiate between gene set enrichment analysis and pathway enrichment analysis...if so, we should keep separate paragraphs in the report; otherwise I can add gprofiler into the existing loop

Reply:
I think pathway is more function-focussed, gene set sounds broader and not as specific to a certain purpose. I also looked the terms up, seems they are indeed (at least sometimes) distinguished, like [here](https://advaitabio.com/science/pathway-analysis-vs-gene-set-analysis/) and [here](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0751-5) (abstract).
But I don't specifically mind either way. However, I just realized that the GMT files may look different for GSEA and gprofiler2, so if we move the two together into one category and e.g. use only a single GMT parameter, I think we might have to keep those separate...

Reply:
So, functionally these things are the same thing. We're talking sets of genes whether you mean pathways, GO terms, whatever- so let's keep to gene set (unless you propose limiting users to 'pathway' gene sets in some way). 

Could you expand on the GMT file differences? I would expect them to be interchangeable, so if they are indeed different we should document what the blocking differences are for each method- e.g. why a gene sets file I used for GSEA couldn't also be used for gprofiler.

Reply:
Fundamentally the database shouldn't matter, these are just 'bags of genes', with identifier types corresponding to the ones you use in your input matrices. The source is irrelevant at run time (unless gprofiler is doing something I don't understand).

Reply:
And no, there's no gene set database invoked at runtime in GSEA as we're using it- we supply the gene sets via the GMT files and GSEA does the stats.

Reply:
I added gprofiler to the GSEA section. I also created a new param group called mods_ into which we can put all parameters that affect multiple modules (I'm open for suggestions for better names) and added both round_digits and gene_sets into that category; I felt like gene_sets does not really fit report_. There is also a short documentation section about GSEA; if you want, you can also send me some more elaborate text and I'll add that in